plugins/modules/packaging/language/npm.py

```diff
@@ -174,7 +174,7 @@ def _exec(self, args, run_in_check_mode=False, check_rc=True):
                 cmd.append('--ignore-scripts')
             if self.unsafe_perm:
                 cmd.append('--unsafe-perm')
-            if self.name:
+            if self.name and add_package_name:
```

Comment:
This change breaks all version specific updates because this was the way npm was checking if a specific version is installed.

Before:
`npm list --json --long pkg@1.2.3` would return `{}` if `pkg` was not installed with version `1.2.3`

After
`npm list --json --long` would return the full `pkg` info regardless of which version is installed, and this will make ansible think that the module is installed.

Reply:
Yes, it looks like you are correct. (Though I wouldn't say that it is this change, but the call of this function with `add_package_name=False`.) So either it passes `add_package_name=False`, or it has special code in `list()` to check for the correct version.

This is also relevant for `state=absent`, since before the module would only uninstall the package if the version matches (assuming the version is provided), while now it will always uninstall it and ignore the version. The new behavior better matches the documentation IMO, but this is still different from before.

@Akasurde can you take a look?

Reply:
@Akasurde ping ^